packages/grid/x-data-grid/src/hooks/features/columns/useGridColumns.tsx

```diff
@@ -179,22 +182,22 @@ export function useGridColumns(
         apiRef.current.forceUpdate();
       }
     },
-    [apiRef, columnsTypes],
+    [apiRef, columnTypes],
   );
 
   const updateColumns = React.useCallback<GridColumnApi['updateColumns']>(
     (columns) => {
       const columnsState = createColumnsState({
         apiRef,
-        columnsTypes,
+        columnsTypes: columnTypes,
```

Comment:
```suggestion
        columnTypes: columnTypes,
```

Not blocking of course, but we can rename it on `createColumnState` as well